service/src/main/resources/db/migration/V262__child_absences_function.sql

```diff
@@ -0,0 +1,136 @@
+CREATE FUNCTION absences_on_date(childId uuid, theDate date)
+    RETURNS TABLE(
+                     id uuid,
+                     child_id uuid,
+                     date date,
+                     absence_type absence_type,
+                     modified_at timestamp with time zone,
+                     modified_by uuid,
+                     category absence_category,
+                     questionnaire_id uuid
+                 )
+AS $$
+BEGIN
+    RETURN QUERY
+        SELECT *
+        FROM absence a
+        WHERE a.child_id = $1 AND a.date = $2
+        UNION
+        SELECT
+            dst.id as id,
+            dst.child_id,
+            $2 as date,
+            'OTHER_ABSENCE'::absence_type as absence_type,
+            dst.updated as modified_at,
+            null::uuid as modified_by,
+            'BILLABLE'::absence_category as category,
+            null::uuid as questionnaire_id
+        FROM daily_service_time as dst
+        WHERE dst.child_id = $1
+          AND dst.validity_period @> $2
+          AND $2 >= lower(dst.validity_period)
+          AND dst.type = 'IRREGULAR'
+          AND (
+                (extract(dow FROM theDate) = 0 AND dst.sunday_times IS NULL)
+                OR (extract(dow FROM theDate) = 1 AND dst.monday_times IS NULL)
+                OR (extract(dow FROM theDate) = 2 AND dst.tuesday_times IS NULL)
+                OR (extract(dow FROM theDate) = 3 AND dst.wednesday_times IS NULL)
+                OR (extract(dow FROM theDate) = 4 AND dst.thursday_times IS NULL)
+                OR (extract(dow FROM theDate) = 5 AND dst.friday_times IS NULL)
+                OR (extract(dow FROM theDate) = 6 AND dst.saturday_times IS NULL)
+            );
+END;
+$$
+    LANGUAGE plpgsql;
+
+CREATE FUNCTION absences_in_range(period daterange)
+    RETURNS TABLE(
+                     id uuid,
+                     child_id uuid,
+                     date date,
+                     absence_type absence_type,
+                     modified_at timestamp with time zone,
+                     modified_by uuid,
+                     category absence_category,
+                     questionnaire_id uuid
+                 )
+AS $$
+BEGIN
+    RETURN QUERY
+        SELECT *
+        FROM absence a
+        WHERE between_start_and_end($1, a.date)
+        UNION
+        SELECT
+            dst.id as id,
+            dst.child_id,
+            d::date as date,
+            'OTHER_ABSENCE'::absence_type as absence_type,
+            dst.updated as modified_at,
+            null::uuid as modified_by,
+            'BILLABLE'::absence_category as category,
+            null::uuid as questionnaire_id
+        FROM daily_service_time as dst,
+             generate_series(lower($1), upper($1), '24 hours') as d
+        WHERE dst.validity_period && $1
+          AND d >= lower(dst.validity_period)
+          AND dst.type = 'IRREGULAR'
+          AND (
+                (extract(dow FROM d) = 0 AND dst.sunday_times IS NULL)
```

Comment:
I think using `CASE … WHEN …` could look cleaner here (and also above), similar to https://github.com/espoon-voltti/evaka/blob/master/service/src/main/kotlin/fi/espoo/evaka/reservations/ReservationQueries.kt#L27-L35.